jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+import jax.numpy as jnp
+import warnings
+from numpy import  ma
+from . import mstats_basic
+import numpy as np
+
+def _chk_asarray(a, axis):
+    if axis is None:
+        a = jnp.ravel(a)
+        outaxis = 0
+    else:
+        a = jnp.asarray(a)
+        outaxis = axis
+
+    if a.ndim == 0:
+        a = jnp.atleast_1d(a)
+
+    return a, outaxis
+
+ModeResult = namedtuple('ModeResult', ('mode', 'count'))
+
+def _contains_nan(a, nan_policy='propagate'):
+    policies = ['propagate', 'raise', 'omit']
+    if nan_policy not in policies:
+        raise ValueError("nan_policy must be one of {%s}" %
+                         ', '.join("'%s'" % s for s in policies))
+    try:
+        # Calling jnp.sum to avoid creating a huge array into memory
+        # e.g. jnp.isnan(a).any()
+        with np.errstate(invalid='ignore'):                                               #check
+            contains_nan = jnp.isnan(jnp.sum(a))
+    except TypeError:
+        # This can happen when attempting to sum things which are not
+        # numbers (e.g. as in the function `mode`). Try an alternative method:
+        try:
+            contains_nan = jnp.nan in set(a.ravel())
+        except TypeError:
+            # Don't know what to do. Fall back to omitting nan values and
+            # issue a warning.
+            contains_nan = False
+            nan_policy = 'omit'
+            warnings.warn("The input array could not be properly "
+                          "checked for nan values. nan values "
+                          "will be ignored.", RuntimeWarning)
+
+    if contains_nan and nan_policy == 'raise':
+        raise ValueError("The input contains nan values")
+
+    return contains_nan, nan_policy
+
+def mode(a, axis=0, nan_policy='propagate'):
```

Comment:
Rather than copying the contents of the docstring from scipy (which has licensing implications), we instead use the `_wraps` decorator to add it automatically. You can look at functions defined in `jax/_src/numpy/lax_numpy.py` to see how it is used.